### PR TITLE
Improved header comments on generated files

### PR DESCRIPTION
**Pull request by darwin-morocho**
Why fix is required?
1. Currently all generated files with `slang` are included in the coverage report after run `flutter test --coverage`. With this PR the all generated files will include the `coverage:ignore-file` comment.

2. Also in the `generate_header.dart` file I have made some little changes to improve the code readability using a multi-line string and replacing 
```
ignore_for_file: prefer_const_constructors, prefer_const_literals_to_create_immutables, library_private_types_in_public_api, unnecessary_string_interpolations, avoid_escaping_inner_quotes
```
with 
```
ignore_for_file: type=lint
```

After run `flutter pub run slang` we have the next result
```dart
/// Generated file. Do not edit.
///
/// Locales: 2
/// Strings: 12 (6 per locale)
///
/// Built on 2022-08-07 at 18:29 UTC


// coverage:ignore-file
// ignore_for_file: type=lint
```

3. Finally the `strings.g.dart` file was removed and ignored for futures pushes because it is a generated file

### CODE REVIEW

File: slang/example/lib/i18n/strings.g.dart [hunk 1]
```diff
@@ -1,247 +0,0 @@
-
```
Comment (Tienisto) [inline slang/example/lib/i18n/strings.g.dart:1]: The generated file should not be deleted because it is referenced by the readme.

**Review by Tienisto — CHANGES_REQUESTED** [review]
Hi,
thanks for your contribution!
I agree with most of your points. However, the generated file is tracked by git on purpose because it is referenced by the readme. Developers interested into this library may like to see how a generated file may look like.

File: slang/example/pubspec.yaml [hunk 2]
```diff
@@ -11,8 +11,10 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  slang: ^2.6.1 # add this
-  slang_flutter: ^2.6.0 # add this
+  slang:
```
[the remark this answers is not in the document]
Comment (darwin-morocho) [reply slang/example/pubspec.yaml]: Solved

File: slang/example/lib/i18n/strings.g.dart (hunk 1 above)
Comment (darwin-morocho) [reply slang/example/lib/i18n/strings.g.dart:1]: Solved

**Comment by Tienisto** [comment]
Thank you!

**Pull request merged**